openedx/core/djangoapps/ccxcon/api.py

```diff
@@ -0,0 +1,156 @@
+        client_id (str): oauth client ID
+        client_secret (str): oauth client secret
+
+    Returns:
+        OAuth2Session: an instance of OAuth2Session with a token
+    """
+    if not is_valid_url(server_token_url):
+        return
+    client = BackendApplicationClient(client_id=client_id)
+    oauth_ccxcon = OAuth2Session(client=client)
+    oauth_ccxcon.fetch_token(
+        token_url=server_token_url,
+        client_id=client_id,
+        client_secret=client_secret
+    )
+    return oauth_ccxcon
+
+
+def course_info_to_ccxcon(course_key):
+    """
+    Function that gathers informations about the course and
+    makes a post request to a CCXCon with the data.
+
+    Args:
+        course_key (CourseLocator): the master course key
+    """
+
+    try:
+        course = get_course_by_id(course_key)
+    except Http404:
+        log.error('Master Course with key "%s" not found', unicode(course_key))
+        return
+    if not course.enable_ccx:
+        log.debug('ccx not enabled for course key "%s"', unicode(course_key))
+        return
+    if not course.ccx_connector:
+        log.debug('ccx connector not defined for course key "%s"', unicode(course_key))
+        return
+    if not is_valid_url(course.ccx_connector):
+        log.error(
+            'ccx connector URL "%s" for course key "%s" is not a valid URL.',
+            course.ccx_connector, unicode(course_key)
+        )
+        return
+    # get the oauth credential for this URL
+    try:
+        ccxcon = CCXCon.objects.get(url=course.ccx_connector)
+    except CCXCon.DoesNotExist:
+        log.error('ccx connector Oauth credentials not configured for URL "%s".', course.ccx_connector)
+        return
+
+    # get an oauth client with a valid token
+
+    oauth_ccxcon = get_oauth_client(
+        server_token_url=urlparse.urljoin(course.ccx_connector, CCXCON_TOKEN_URL),
+        client_id=ccxcon.oauth_client_id,
+        client_secret=ccxcon.oauth_client_secret
+    )
+
+    # get the entire list of instructors
+    course_instructors = list_with_level(course, 'instructor')
+    # get anonymous ids for each of them
+    course_instructors_ids = [anonymous_id_for_user(user, course_key) for user in course_instructors]
+    # extract the course details
+    course_details = CourseDetails.fetch(course_key)
+
+    payload = {
+        'course_id': unicode(course_key),
+        'title': course.display_name,
+        'author_name': None,
+        'overview': course_details.overview,
+        'description': course_details.short_description,
+        'image_url': course_details.course_image_asset_path,
+        'instructors': course_instructors_ids
+    }
+    headers = {'content-type': 'application/json'}
+
+    # make the POST request
+    add_course_url = urlparse.urljoin(course.ccx_connector, CCXCON_COURSEXS_URL)
+    resp = oauth_ccxcon.post(
+        url=add_course_url,
+        json=payload,
+        headers=headers
+    )
```

Comment:
I thought it did, but it does not.
`oauth_ccxcon.post` is an instance of `OAuth2Session` that combines `OAuth2` and `requests`. The post comes from the `requests` part and the default is `None`. 
[link to docs](http://docs.python-requests.org/en/latest/api/#requests.Session.request)

I will change this.